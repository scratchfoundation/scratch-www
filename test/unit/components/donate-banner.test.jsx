import React from 'react';
import {mountWithIntl} from '../../helpers/intl-helpers.jsx';
import DonateTopBanner from '../../../src/views/splash/donate/donate-banner';

describe('DonateBannerTest', () => {
    let realDateNow;

    beforeEach(() => {
        realDateNow = Date.now.bind(global.Date);
    });
    afterEach(() => {
        global.Date.now = realDateNow;
    });
    test('Testing 2024 EOY campaign message', () => {
        global.Date.now = () => new Date(2024, 11, 16).getTime();
        const component = mountWithIntl(
            <DonateTopBanner />
        );

        expect(component.find('div.donate-banner').exists()).toEqual(true);
        expect(component.find('p.donate-text').exists()).toEqual(true);
        expect(component.find('FormattedMessage[id="donatebanner.eoyCampaign"]').exists()).toEqual(true);
        expect(component.find('FormattedMessage[id="donatebanner.askSupport"]').exists()).toEqual(false);

    });
    test('testing default message comes back after January 9, 2025', () => {
        // Date after Scratch week
        global.Date.now = () => new Date(2025, 0, 10).getTime();
        const component = mountWithIntl(
            <DonateTopBanner />
        );
        expect(component.find('div.donate-banner').exists()).toEqual(true);
        expect(component.find('p.donate-text').exists()).toEqual(true);
        expect(component.find('FormattedMessage[id="donatebanner.askSupport"]').exists()).toEqual(true);
        expect(component.find('FormattedMessage[id="donatebanner.eoyCampaign"]').exists()).toEqual(false);
    });
});
