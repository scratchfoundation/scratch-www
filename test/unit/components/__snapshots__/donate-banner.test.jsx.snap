// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonateBannerTest testing default message 1`] = `
<div
  class="title-banner donate-banner"
>
  <div
    class="donate-container"
  >
    <img
      aria-hidden="true"
      class="donate-icon"
      src="/images/ideas/try-it-icon.svg"
    />
    <div
      class="donate-central-items"
    >
      <p
        class="donate-text"
      >
        Scratch is a nonprofit that relies on donations to keep our platform free for all kids. Your gift of $5 will make a difference.
      </p>
      <button
        class="button donate-button"
      >
        Donate
      </button>
    </div>
  </div>
  <button
    class="button donate-close-button forms-close-button"
    name="closeButton"
    type="button"
  >
    <img
      alt="close-icon"
      class="modal-content-close-img"
      draggable="false"
      src="/svgs/modal/close-x.svg"
    />
  </button>
</div>
`;
